rdrf/rdrf/file_upload.py

```diff
@@ -62,9 +66,38 @@ def wrap(value, key):
             return [wrap(item, key) for item in value]
         elif filestorage.get_id(value):
             return FileUpload(registry, key, value)
+        elif isinstance(value, UploadedFile):
+            return FileUpload(registry, key,
+                              {"file_name": value.name, "django_file_id": 0})
         elif isinstance(value, dict):
-            for key, item in value.items():
-                value[key] = wrap(item, key)
+            return { key: wrap(item, key) for key, item in value.items() }
         return value
 
     return wrap(data, None)
+
+
+def merge_gridfs_data_for_form(registry, form_data, dyn_patient):
+    """
+    Merges existing patient mongo from dyn_patient into form_data, if
+    it's missing.
+    Because form_data is cleaned_data from a Django form, file uploads
+    are missing. In order to display the currently existing files,
+    they need to be copied over from dyn_patient.
```

Comment:
Yes I thought this was the problem I caused after changing the form section validation to all or nothing